clang/lib/Sema/SemaChecking.cpp

```diff
@@ -1350,10 +1360,17 @@ void Sema::checkFortifiedBuiltinMemoryFunction(FunctionDecl *FD,
             llvm::APSInt::getUnsigned(H.getSizeLowerBound())
                 .extOrTrunc(SizeTypeWidth);
         if (FormatSize > *SourceSize && *SourceSize != 0) {
-          DiagID = diag::warn_fortify_source_format_truncation;
-          DestinationSize = SourceSize;
-          SourceSize = FormatSize;
-          break;
```

Comment:
This diagnostic used to override the production of the `warn_fortify_source_size_mismatch` warning, and now is produced in addition to it. Is that intentional?

Reply:
This is intentional.
When the truncation warning was grouped under `Wfortify-source`, I thought it made sense to override the warning because there was no way to disable only one of `warn_fortify_source_size_mismatch` and the truncation warning.
Now that `Wformat-truncation` is made, I no longer think this warning should hide the `Wfortify-source` warning.